### PR TITLE
[Server] WMS GetStyles (and GetStyle) refactoring

### PR DESCRIPTION
**Pull request by rldhont**
Removing QgsWMSProjectParser from GetStyles and GetStyle
https://github.com/qgis/qgis3.0_api/issues/57

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/server/services/wms/qgswmsgetstyles.cpp [hunk 1]
```diff
@@ -50,14 +58,137 @@ namespace QgsWms
                                     QStringLiteral( "Layers is mandatory for GetStyles operation" ) );
     }
 
-    QStringList layersList = layersName.split( QStringLiteral( "," ), QString::SkipEmptyParts );
-    if ( layersList.size() < 1 )
+    QStringList layerList = layersName.split( QStringLiteral( "," ), QString::SkipEmptyParts );
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsgetstyles.cpp]: .split(',') is much faster - in general any qstring operation which can be replaced with a character based version will be much more performant

File: src/server/services/wms/qgswmsgetstyles.cpp [hunk 2]
```diff
@@ -50,14 +58,137 @@ namespace QgsWms
...
+    if ( layerList.size() < 1 )
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsgetstyles.cpp]: if ( layerList.isEmpty() )

(easier to immediately understand)

**Review by nyalldawson — COMMENTED** [review]
Just some small code suggestions

File: src/server/services/wms/qgswmsgetstyles.cpp (hunk 1 above)
Comment (rldhont) [inline src/server/services/wms/qgswmsgetstyles.cpp]: thanks, we have to review all the server code.

File: src/server/services/wms/qgswmsgetstyles.cpp (hunk 2 above)
Comment (rldhont) [inline src/server/services/wms/qgswmsgetstyles.cpp]: thanks, we have to review all the server code.

**Pull request merged**